### PR TITLE
Feature/esther/cns 22 internet file service upload file

### PR DESCRIPTION
**Pull request by Esther125**
## Changes
- Implement file upload function.

## Tests
- Refer to `Project SetUp and Start` part in README.md to start the server.
- Download [Postman](https://www.postman.com/downloads/)
- Setting up Postman as the picture below and upload any file to test the API endpoint. You will get a unique fileId if success. You can also check if the file you uploaded is really saved under the path: `backend/uploads`
image

### CODE REVIEW

**Review by yun-si — CHANGES_REQUESTED** [review]
如果沒有檔案上傳，會回傳 400 沒錯
但 app 會 crash
但應該是要避免 app crash 的

截圖 2024-11-10 晚上10 25 38

**Comment by yun-si** [comment]
上面的問題解決了！
只是還有一些 coding style 的部分，例如：使用箭頭函數，還有 conflict 要解一下～

```
async upload(req, res) {}
```
->
```
upload = async  (req, res) => {}
```

File: backend/src/services/internetFileService.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+    async upload(req, res) {
+        // 實現上傳檔案邏輯
+        if (!req.files || Object.keys(req.files).length === 0) {
+            throw new Error('No files were uploaded.');
+        }
+
+        const uploadFile = req.files.uploadFile;
+
+        const __filename = fileURLToPath(import.meta.url);
```
Comment (yun-si) [inline backend/src/services/internetFileService.js:20]: 想問問：為什麼這裡會用兩個底線開頭，命名這個變數

Comment (Esther125) [reply backend/src/services/internetFileService.js:20]: 這是 node.js 的習慣，那兩個變數算是比較特別的變數

File: backend/src/controllers/internetFileController.js [hunk 2]
```diff
@@ -1,15 +1,30 @@
...
+
 class InternetFileController {
-    async upload(req, res) {
-        console.log('----InternetFileController.upload');
-        // TODO: 實現上傳檔案邏輯
-        res.status(201).json({ message: 'File upload logic not implemented yet' });
+    constructor() {
+        this.internetFileService = new InternetFileService();
+        this.upload = this.upload.bind(this);
+        this.download = this.download.bind(this);
```
Comment (yun-si) [inline backend/src/controllers/internetFileController.js]: 用了箭頭函數就不需要這兩行了

**Review by yun-si — APPROVED** [review]
lgtm~~

**Pull request merged**